gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/ObservationUtil.java

```diff
@@ -0,0 +1,213 @@
+        if (effectiveTime != null) {
+            if (effectiveTimeHasCenter(effectiveTime)) {
+                return DateFormatUtil.parseToDateTimeType(effectiveTime.getCenter().getValue());
+            }
+
+            var effectivePeriod = new Period();
+
+            if (effectiveTimeHasLow(effectiveTime)) {
+                effectivePeriod.setStartElement(DateFormatUtil.parseToDateTimeType(effectiveTime.getLow().getValue()));
+            }
+
+            if (effectiveTimeHasHigh(effectiveTime)) {
+                effectivePeriod.setEndElement(DateFormatUtil.parseToDateTimeType(effectiveTime.getHigh().getValue()));
+            }
+
+            if (availabilityTimeHasValue(availabilityTime)) {
+                if (effectivePeriod.getStart() == null) {
+                    if (effectivePeriod.getEnd() != null) {
+                        effectivePeriod.setStartElement(DateFormatUtil.parseToDateTimeType(availabilityTime.getValue()));
+                    }
+                }
+            }
+
+            return effectivePeriod;
+        } else if (availabilityTimeHasValue(availabilityTime)) {
+            return DateFormatUtil.parseToDateTimeType(availabilityTime.getValue());
+        }
+        return null;
+    }
+
+    private static boolean isValidValueQuantity(Object value) {
+        return value instanceof PQ || value instanceof IVLPQ;
+    }
+
+    private static String getInterpretationCodeAbbreviation(String interpretationCode) {
+        return switch (interpretationCode) {
+            case ("HI") -> "H";
+            case ("LO") -> "L";
+            case ("OR") -> "A";
+            default -> StringUtils.EMPTY;
+        };
+    }
+
+    private static String getInterpretationDisplay(String interpretationCode) {
+        return switch (interpretationCode) {
+            case ("HI") -> "High";
+            case ("LO") -> "Low";
+            case ("OR") -> "Abnormal";
+            default -> StringUtils.EMPTY;
+        };
+    }
+
+    private static boolean referenceInterpretationRangeHasValue(RCMRMT030101UK04InterpretationRange referenceInterpretationRange) {
+        return referenceInterpretationRange != null && referenceInterpretationRange.getValue() != null;
+    }
+
+    private static SimpleQuantity getSimpleQuantityFromQuantity(Quantity quantity, long newValue) {
+        return (SimpleQuantity) new SimpleQuantity()
+            .setValue(newValue)
+            .setUnit(quantity.getUnit())
+            .setCode(quantity.getCode())
+            .setSystem(quantity.getSystem())
+            .setComparator(quantity.getComparator());
+    }
+
+    private static boolean authorHasValidTimeValue(RCMRMT030101UK04Author author) {
+        return author != null && author.getTime() != null
+            && author.getTime().getValue() != null
+            && author.getTime().getNullFlavor() == null;
+    }
+
+    private static boolean timeHasNullFlavor(TS time) {
+        return time.getNullFlavor() != null;
```

Comment:
I think the name of this method doesn't match the action - you're checking if it hasn't got a null flavour :D Also, you always use this method with `!`, so probably this method should just check if `time.getNullFlavor() == null`